javabridge/locate.py

```diff
@@ -121,28 +127,36 @@ def get_out(cmd):
         jdk_dir = os.path.abspath(jdk_dir)
         return jdk_dir
     elif is_win:
-        if sys.version_info.major == 2:
-            import _winreg as winreg
-        else:
-            import winreg
-        java_key_path = 'SOFTWARE\\JavaSoft\\Java Runtime Environment'
-        looking_for = java_key_path
-        try:
-            kjava = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, java_key_path)
-            looking_for = java_key_path + "\\CurrentVersion"
-            kjava_values = dict([winreg.EnumValue(kjava, i)[:2]
-                                 for i in range(winreg.QueryInfoKey(kjava)[1])])
-            current_version = kjava_values['CurrentVersion']
-            looking_for = java_key_path + '\\' + current_version
-            kjava_current = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE,
-                                            looking_for)
-            kjava_current_values = dict([winreg.EnumValue(kjava_current, i)[:2]
-                                         for i in range(winreg.QueryInfoKey(kjava_current)[1])])
-            return kjava_current_values['JavaHome']
-        except:
-            logger.error("Failed to find registry entry: %s\n" %looking_for,
-                         exc_info=True)
-            return None
+        # Registry keys changed in 1.9
+        # https://docs.oracle.com/javase/9/migrate/toc.htm#GUID-EEED398E-AE37-4D12-AB10-49F82F720027
+        java_key_paths = (
+            'SOFTWARE\\JavaSoft\\JRE',
+            'SOFTWARE\\JavaSoft\\Java Runtime Environment',
+        )
+        for java_key_path in java_key_paths:
+            looking_for = java_key_path
+            try:
+                kjava = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, java_key_path)
+                looking_for = java_key_path + "\\CurrentVersion"
+                kjava_values = dict([winreg.EnumValue(kjava, i)[:2]
+                                     for i in range(winreg.QueryInfoKey(kjava)[1])])
+                current_version = kjava_values['CurrentVersion']
+                looking_for = java_key_path + '\\' + current_version
+                kjava_current = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE,
+                                                looking_for)
+                kjava_current_values = dict([winreg.EnumValue(kjava_current, i)[:2]
+                                             for i in range(winreg.QueryInfoKey(kjava_current)[1])])
+                return kjava_current_values['JavaHome']
+            except WindowsError as e:
+                if e.errno == 2:
+                    continue
+                else:
+                    raise
+
+        raise RuntimeError(
+            "Failed to find the Java Runtime Environment. "
+            "Please download and install the Oracle JRE or later"
```

Comment:
@LeeKamentsky do you have a minimum JRE version that should be specified?  The JDK search below requests at least 1.6.